corehq/apps/userreports/tasks.py

```diff
@@ -80,15 +83,22 @@ def rebuild_indicators(indicator_config_id):
         redis_key = _get_redis_key_for_config(config)
 
     adapter.rebuild_table()
-    relevant_ids = get_doc_ids_in_domain_by_type(
-        config.domain,
-        config.referenced_doc_type,
-        database=couchdb,
-    )
-    for docs in chunked(relevant_ids, 1000):
-        redis_client.sadd(redis_key, *docs)
-
-    _build_indicators(indicator_config_id, relevant_ids)
+    view_kwargs = {
+        'reduce': False,
+        'startkey': [config.domain, config.referenced_doc_type],
+        'include_docs': False
+    }
+    relevant_ids_chunk = []
+    for relevant_id in paginate_view(
+            couchdb,
+            'by_domain_doc_type_date/view',
```

Comment:
yeah so i think my assumption was that this kind of operation was idempotent. skipping newly added docs or reprocessing one or two ids is OK. but let me know if you think that assumption breaks down